classes/setup_form.class.php

```diff
@@ -206,26 +206,29 @@ public function display_features() {
 
         // Display Search Repositories.
         $repolist = html_writer::tag('dt', get_string('turnitinfeatures::repositories', 'plagiarism_turnitinsim'));
-        foreach ($features['similarity']['generation_settings']['search_repositories'] as $repo) {
-            $repolist .= html_writer::tag('dd', ucwords(strtolower(str_replace('_', ' ', $repo))));
-        }
-        $enabledfeatures .= html_writer::tag('dl', $repolist, array('class' => 'turnitinsim_featurelist'));
-
-        // Display View Options.
-        $settinglist = html_writer::tag('dt', get_string('turnitinfeatures::viewoptions', 'plagiarism_turnitinsim'));
-        foreach ($features['similarity']['view_settings'] as $setting => $enabled) {
-            $icon = ($enabled) ? 'option-yes' : 'option-no';
-            $icon = $OUTPUT->pix_icon($icon, '', 'plagiarism_turnitinsim', array('class' => 'turnitinsim_optionicon'));
-            $settinglist .= html_writer::tag('dd', $icon.ucwords(str_replace('_', ' ', $setting)));
-        }
-        $enabledfeatures .= html_writer::tag('dl', $settinglist, array('class' => 'turnitinsim_featurelist'));
 
-        $eulastring = (!(bool)$features['tenant']['require_eula']) ? 'eulanotrequired' : 'eularequired';
-        $enabledfeatures .= html_writer::tag(
-            'p',
-            get_string('turnitinfeatures::'.$eulastring, 'plagiarism_turnitinsim'),
-            array('class' => 'bold')
-        );
+        if (!empty($features)) {
```

Comment:
Do you need to check for it being an empty JSON string `{}` as well?

Reply:
Makes sense to yeah, in case the call to get them returns the same again.